### ISSUE TITLE
Library completely broken since 0.10.0 on some devices

### ISSUE THREAD

**Issue by alaegin** [deepmedia/Transcoder#138]
Hello! 
We have just started to use this library and realized that it's completely broken on some devices since 0.10.0.

Prerequisites:
- [CameraView](https://github.com/natario1/CameraView) library for video recording. Default settings with Camera2 Engine.
- OnePlus 5 and OnePlus 7 Pro. Both use Android 10. _(Also tested on Honor 9X (STK-LX1) and it works fine there)._

The main problem is that the library crashes unexpectedly in some cases or transcode video with artifacts.
- 0.9.1 works fine
- 0.10.0 - 0.10.2 print `Validator has decided that the input is fine and transcoding is not necessary.` and don't transcode video at all. [Link](https://pastebin.com/Qn3JGKdV)
- 0.10.3 fails with several random issues listed below

We have several cases:
1. Library crashes with exception `java.lang.IllegalStateException: Failed to stop the muxer`. [Link](https://pastebin.com/4QAv0T95)
2. Library crashes with exception `TranscodeEngine: Unexpected error while transcoding. java.nio.BufferOverflowException`. [Link](https://pastebin.com/eAecxdEW)
3. Library transcode video with artifacts. [Link to video](https://user-images.githubusercontent.com/10329086/113989045-b835a700-9858-11eb-93c9-19e55b4b5d42.mp4)

The code we use to call Transcoder:

```kotlin
Transcoder.into(filesProvider.getTempResultFile().absolutePath)
    .apply {
        files.forEach { addDataSource(it.absolutePath) }
    }
    .setListener(...)
    .transcode().get()
```

It seems that rewriting #118 of the transcoder pipeline broke something important.

**Comment by natario1** [comment]
Can you provide the media files to reproduce these issues in the demo app?

**Comment by alaegin** [comment]
Videos that were converted with artifacts on OnePlus phones but without them on the android emulator:
[Part #1](https://user-images.githubusercontent.com/10329086/114008100-95ac8980-986a-11eb-9c2f-9d0a14f8b65b.mp4)
[Part #2](https://user-images.githubusercontent.com/10329086/114008331-d0aebd00-986a-11eb-9e21-3bbcd4a09f2e.mp4)

Videos from OnePlus 5
[Part #1](https://user-images.githubusercontent.com/10329086/114008733-34d18100-986b-11eb-8813-3fc4c2294183.mp4)
[Part #2](https://user-images.githubusercontent.com/10329086/114008871-50d52280-986b-11eb-9d70-8f9d6a03d756.mp4)

**Comment by jumperson** [comment]
> Library transcode video with artifacts. Link to video

I have same issue in below devices.

- [F-02L](https://www.fmworld.net/product/phone/f-02l/spec.html) (OS: 9)
- [Lenovo Tab P11 Pro](https://www.lenovo.com/jp/ja/tablets/android-tablets/p-series/Lenovo-TB-J706/p/ZZITZTATB0J) (OS: 10)

**Comment by natario1** [comment]
IMO these issues should be fixed by latest PRs - #159, #200, #201 . Another issue that remains related to buffer overflow is tracked at #163 .
